src/main.js

```diff
@@ -244,3 +244,81 @@ let destinations = [
   ]
 
 
+  //visited and bucktelist lists
+  const visited = destinations.filter(item => item.visited === true);
+  const bucketlist = destinations.filter(item => item.visited === false);
+ 
+
+  //get list of urls
+  const getURL = (destinations) => {
+    let url= destinations.map(obj => obj.photo.valueOf());
+    return url;
+  }
+
+  //create elements for each city
+  const elements= destinations.forEach(destination => {
+
+    //picture element
+    const createImg = (url) => {
+      let figure = document.createElement('figure');
+    //img element
+      let img = document.createElement('img');
+      img.style.objectFit ='cover';
+      document.body.appendChild(figure);
+      figure.appendChild(img);
+    //img caption
+      let figcaption= document.createElement('figcaption');
+      figure.appendChild(figcaption);
+      figcaption.innerText= `${destinations.city}, ${destinations.country}`
+
+
+      //get urls
+      for(let i=0; i<destinations.length; i++) {
+        img.src = url[i];
+      };
+      
+      //add classes 
+      if(destination.visited === true) {
+        img.classList.add('visited');
+      }
+      else {
+        img.classList.add('bucketlist');
+      }
+
+  };
+
+
+
+  return createImg(getURL(destinations));
+
+});
+
+  //create buttons
+  const buttonVisited = document.createElement('button');
+  buttonVisited.innerText='Visited';
+  buttonVisited.type='button';
+  document.body.appendChild(buttonVisited);
+  
+  
+  const buttonBucketlist = document.createElement('button');
+  buttonBucketlist.innerText='Bucketlist';
+  buttonBucketlist.type='button';
+  document.body.appendChild(buttonBucketlist);
+  
+
+//connect lists with buttons
+
+buttonVisited.addEventListener('click', () => {
+  document.querySelectorAll('.visited');
```

Comment:
To do this, instead of your approach (that is complex), it's easier if we add a class to the list parent and then with CSS you manipulate the list items based on that class. For example:

```html
<div class="list">
  <figure class="visited">...</figure>
  <figure class="bucketList">...</figure>
  <figure class="visited">...</figure>
</div>
```
// with JS add a class ".visited" to list when clicking the button. so The DOM is like this:
```html
<div class="list active">
  ....
</div>
```

... and with CSS you hide the elements based on the list class:
```css
.list.visited figure {
  display: none; // Hide all items (figure) when .list has .visited class
} 

.list.active figure.visited {
  display: block; // overide the figure.visited items to be visible
}
```

This is where CSS selectors real powers come into action! Good luck!

Reply:
It does sound much less complicated, I'll redo  it. This is the code I forgot to push, I'm curious why I wasn't able to select the .visited sibiling (two ways I tried are on lines 3-5) 

```buttonVisited.addEventListener('click', () => {
  const visited = document.querySelectorAll('.visited');
  // const captions = document.querySelectorAll('.visited .caption');   
  const captions = visited.forEach(item => {
    return item.nextElementSibling;
  })

  //add display:none to each img and figcaption 
  visited.forEach(item => {
    item.style.display='none';
  })
  captions.forEach(item => {
    item.style.display='none';
  });

});
```

Reply:
This is because `forEach` is being used incorrectly. I recommend you re-reading again the [diff between `forEach()` and `map()`](https://codeburst.io/javascript-map-vs-foreach-f38111822c0f). In this case, you can use both. Personally I'd use map because I like [declarative paradigm](https://dzone.com/articles/imperative-vs-declarative-javascript#:~:text=Imperative%20and%20Declarative%20Styles,is%20that%20you%20want%20done.).

I challenge you to try come up with both approaches (forEach and map) in this scenario to get the `.captions`.Try it, before I share the solution ;)

Reply:
Just saw this now after moving away from that solution but trying map with my current one didn't work (because it returns a new array and I need the unaltered one..?)

I wasn't really getting this part 
```
// with JS add a class ".visited" to list when clicking the button. so The DOM is like this:

<div class="list active">
  ....
</div>
```
so I tried figuring it out on my own and it worked! What I don't like about my it is that I had to copy the same variables in each block, is there a better way to access them?

Reply:
> (because it returns a new array and I need the unaltered one..?)

It returns a new array but you still have access to the original one (it isn't mutated)

> What I don't like about it is that I had to copy the same variables in each block, is there a better way to access them?

I think you didn't understand it fully. You just need to add `. visited ` or `.bucketlist` to the items list parent (`.gallery`), not to each list item. That way the code will be much smaller.

And then, with CSS you do the display block/none as I showed in the snippet above.


I'll share a solution soon, don't worry. Just give it a try again based on my new indications. ;)

Reply:
Still confused about that snippet, why ```.active``` here?
```
.list.active figure.visited {
  display: block; // overide the figure.visited items to be visible
}
```

Reply:
It was not `.active`, it was `.visited`. I did a small demo showing this filter with css technique. I hope it helps you to understand how to apply it in this project ;)

https://codepen.io/sandrina-p/pen/OJNQJbz